repository_service_tuf/cli/admin/import_targets.py

```diff
@@ -0,0 +1,201 @@
+    request_server,
+    task_status,
+)
+from repository_service_tuf.helpers.tuf import (
+    Delegations,
+    Metadata,
+    SuccinctRoles,
+    Targets,
+)
+
+console = Console()
+
+
+def _check_csv_files(csv_files: List[str]):
+    not_found_csv_files: List[str] = []
+    for csv_file in csv_files:
+        if not os.path.isfile(csv_file):
+            not_found_csv_files.append(csv_file)
+
+    if len(not_found_csv_files) > 0:
+        raise click.ClickException(
+            f"CSV file(s) not found: {(', ').join(not_found_csv_files)}"
+        )
+
+
+def _parse_csv_data(
+    csv_file: str, succinct_roles: SuccinctRoles
+) -> List[Dict[str, Any]]:
+    rstuf_db_data: List[Dict[str, Any]] = []
+    with open(csv_file, "r") as f:
+        for line in f:
+            rstuf_db_data.append(
+                {
+                    "path": line.split(";")[0],
+                    "info": {
+                        "length": int(line.split(";")[1]),
+                        "hashes": {line.split(";")[2]: line.split(";")[3]},
+                    },
+                    "rolename": succinct_roles.get_role_for_target(
+                        line.split(";")[0]
+                    ),
+                    "published": False,
+                    "action": "ADD",
+                    "last_update": datetime.now(),
+                }
+            )
+
+    return rstuf_db_data
+
+
+def _import_csv_to_rstuf(
+    db_client: Connection,
+    rstuf_table: Table,
+    csv_files: List[str],
+    succinct_roles: SuccinctRoles,
+) -> None:
+    for csv_file in csv_files:
+        console.print(f"Import status: Loading data from {csv_file}")
+        rstuf_db_data = _parse_csv_data(csv_file, succinct_roles)
+        console.print(f"Import status: Importing {csv_file} data")
+        try:
+            db_client.execute(rstuf_table.insert(), rstuf_db_data)
+        except IntegrityError:
+            raise click.ClickException(
+                "Import status: ABORTED due duplicated targets. "
+                "CSV files must to have unique targets (path). "
+                "No data added to RSTUF DB."
+            )
+        console.print(f"Import status: {csv_file} imported")
+
+
+def _get_succinct_roles(metadata_url: str) -> SuccinctRoles:
+    response = request_server(metadata_url, "1.bin.json", Methods.get)
+    if response.status_code == 404:
+        raise click.ClickException("RSTUF Metadata Targets not found.")
+
+    json_data = json.loads(response.text)
+    targets: Metadata[Targets] = Metadata.from_dict(json_data)
+    if targets.signed.delegations is None:
+        raise click.ClickException("Failed to get Targets Delegations")
+
+    targets_delegations: Delegations = targets.signed.delegations
+
+    if targets_delegations.succinct_roles is None:
+        raise click.ClickException("Failed to get Targets succinct roles")
+
+    return targets_delegations.succinct_roles
+
+
+@admin.command()
+@click.option(
+    "-metadata-url",
```

Comment:
Why here, in `-db-uri` and `-csv` do we not follow the same double dash pattern (`--`) that we do for `--skip-publish-targets`, since they are all multi-phrase options?

Reply:
It was decided in the [Feature PR](https://github.com/vmware/repository-service-tuf/blob/feature/import-targets-to-db/tests/features/targets/import_targets.feature).
But I changed it.
I will open a PR to update the Feature file.

Reply:
I just thought you used a single `-` when it's a parameter with a value and `--` when it's a flag.
Isn't that the style for CLI tools usually?

Reply:
There are no *written* rules.
But what is more common is `-m/--multiple-word-parameters`. We can do it and make it easier to follow this rule in CLI. The short parameter we are not adding but adding in the future does not impact the current implementation. It works more as an alias.

Reply:
I am okay with using `-` for short one-word parameters and `--` for multi-word parameters.
By the way: if we decide to make changes and to use `--` for any of the parameters we need to update the documentation in the `docs/source/guide/index.rst` as well.

Reply:
Updated.